rir/src/compiler/opt/type_speculation.cpp

```diff
@@ -45,13 +45,18 @@ void TypeSpeculation::apply(RirCompiler&, ClosureVersion* function,
                     if (auto ld = LdVarSuper::Cast(arg))
                         if (!Env::isPirEnv(ld->env()))
                             trigger = true;
-                    if (i->arg(0).val()->typeFeedback.orPromiseWrapped() ==
-                            i->typeFeedback.orPromiseWrapped() &&
-                        Instruction::Cast(i->arg(0).val()))
-                        i = Instruction::Cast(i->arg(0).val());
-                    break;
-                }
-                default: {}
+                    if (force->observed != Force::ArgumentKind::unknown &&
+                        force->observed != Force::ArgumentKind::promise &&
+                        Instruction::Cast(force->input())) {
+                        i = Instruction::Cast(force->input());
+                        if (force->observed == Force::ArgumentKind::value)
+                            i->typeFeedback = force->typeFeedback;
```

Comment:
drive by comment: wouldn't this do the exact same thing as the previous try, just a bit later? if multiple force have the same input then this still leads to wrong typefeeback.